pkg/users/account-details.js

```diff
@@ -101,12 +101,23 @@ export function AccountDetails({ accounts, groups, current_user, user, shells })
     useEffect(() => {
         get_expire(user).then(setExpiration);
 
-        // Watch `/var/run/utmp` to register when user logs in or out
-        const handle = cockpit.file("/var/run/utmp", { superuser: "try", binary: true });
-        handle.watch(() => {
-            get_expire(user).then(setExpiration);
-        }, { read: false });
-        return handle.close;
+        const lastlog_version = cockpit.spawn(["test", "-e", "/var/run/utmp"], { binary: true, err: true });
+
+        lastlog_version.then(() => {
+            // Watch `/var/run/utmp` to register when user logs in or out
+            const handle = cockpit.file("/var/run/utmp", { superuser: "try", binary: true });
+            handle.watch(() => {
+                get_expire(user).then(setExpiration);
+            }, { read: false });
+            return handle.close;
+        }).catch(() => {
+            // assume that we are on lastlog2 and watch /var/lib/lastlog/lastlog2.db instead of /var/run/utmp
+            const handle = cockpit.file("/var/lib/lastlog/lastlog2.db", { superuser: "try", binary: true });
+            handle.watch(() => {
+                get_expire(user).then(setExpiration);
+            }, { read: false });
+            return handle.close;
```

Comment:
The then and catch code is exactly the same apart from th path. If you pass the path as a property, then this will solve itself.